manifests/params.pp

```diff
@@ -20,42 +20,27 @@
   $package_ensure   = 'present'
   $package_name     = 'hiera'
   $hierarchy        = []
-  if str2bool($facts['is_pe']) {
-    $hiera_yaml     = '/etc/puppetlabs/puppet/hiera.yaml'
-    $datadir        = '/etc/puppetlabs/puppet/hieradata'
-    $owner          = 'pe-puppet'
-    $group          = 'pe-puppet'
-    $cmdpath        = ['/opt/puppet/bin', '/usr/bin', '/usr/local/bin']
-    $manage_package = false
-
-    if $::pe_version and versioncmp($::pe_version, '3.7.0') >= 0 {
-      $provider       = 'pe_puppetserver_gem'
-      $master_service = 'pe-puppetserver'
-    } else {
-      $provider       = 'pe_gem'
-      $master_service = 'pe-httpd'
-    }
-  } else {
-    # Configure for AIO packaging.
-    if getvar('::pe_server_version') {
-      $master_service = 'pe-puppetserver'
-      $provider       = 'puppetserver_gem'
-    } else {
-      # It would probably be better to assume this is puppetserver, but that
-      # would be a backwards-incompatible change.
-      $master_service = 'puppetmaster'
-      $provider       = 'puppet_gem'
-    }
-    $cmdpath        = ['/opt/puppetlabs/puppet/bin', '/usr/bin', '/usr/local/bin']
-    $datadir        = '/etc/puppetlabs/code/environments/%{::environment}/hieradata'
-    $manage_package = false
-    if getvar('::pe_server_version') {
-      $owner = 'pe-puppet'
-      $group = 'pe-puppet'
-    } else {
-      $owner = 'puppet'
-      $group = 'puppet'
-    }
-    $hiera_yaml = "${confdir}/hiera.yaml"
+  # Configure for AIO packaging.
+  if getvar('::pe_server_version') {
+    $master_service = 'pe-puppetserver'
+    $provider       = 'puppetserver_gem'
+    $owner          = 'root'
+    $group          = 'root'
+    $eyaml_owner    = 'pe-puppet'
+    $eyaml_group    = 'pe-puppet'
   }
+  else {
+    # It would probably be better to assume this is puppetserver, but that
+    # would be a backwards-incompatible change.
```

Comment:
We're about to make one anyway when we drop support for puppet 4.  Maybe this is a good time?

Reply:
I'm not opposed to it, but I'd really like to get a new version with this fix in immediately for the PE issue (it's restarting pe-puppetserver every 30 minutes because of this) and not worry about the backwards-incompatibility at the same time.

Reply:
ok.  I'll call off [3.4.1](https://github.com/voxpupuli/puppet-hiera/pull/259) and we can do a 3.5.0 instead.

Reply:
Also, I should remember to use 'hide whitespace changes' more often to make it clear to me that this isn't something you had written...